models/Asset.php

```diff
@@ -966,77 +966,107 @@ public function delete(bool $isNested = false): void
             throw new Exception('root-node cannot be deleted');
         }
 
-        $this->dispatchEvent(new AssetEvent($this), AssetEvents::PRE_DELETE);
-
-        $this->beginTransaction();
+        if (!$isNested) {
+            $deletedPaths = [];
+            $deletedIds = [];
+        }
 
         try {
-            $this->closeStream();
+            $this->dispatchEvent(new AssetEvent($this), AssetEvents::PRE_DELETE);
 
-            // remove children
-            if ($this->hasChildren()) {
-                foreach ($this->getChildren() as $child) {
-                    $child->delete(true);
-                }
-            }
+            $maxRetries = 5;
+            for ($retries = 0; $retries < $maxRetries; $retries++) {
+                $this->beginTransaction();
 
-            // Dispatch Symfony Message Bus to delete versions
-            Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
-                new VersionDeleteMessage(Service::getElementType($this), $this->getId())
-            );
+                try {
+                    $this->closeStream();
+
+                    // remove children
+                    if ($this->hasChildren()) {
+                        foreach ($this->getChildren() as $child) {
+                            $child->delete(true);
+                            $deletedPaths[] = $child->getRealPath() . $child->getId();
+                            $deletedIds[] = $child->getId();
```

Comment:
Should this be wrapped in a condition?

```suggestion
                            if (!$isNested) {
                                $deletedPaths[] = $child->getRealPath() . $child->getId();
                                $deletedIds[] = $child->getId();
                            }
```

Reply:
mhhh, it needs somehow to recursively collecting the ids to pass for the main call, if i apply the suggestion, only the direct children would be considered, right?

Reply:
I don't know. I just assumed it should also be in the condition because all other places where these variables are used are.